### PR TITLE
Adding classes BiasedTranslateRotate and ChargeTransfer

### PR DESCRIPTION
**Pull request by vaspelin**
Please check that:

- `make all pyfaunus usagetips unittests test` passes with no errors
- there are no compiler warnings
- no temporary files (.o, .pqr, state etc) are included in the commit
- naming policies for classes, functions, and variables adhere to:

  Style      | Elements
  ---------- | -------------------------
  PascalCase | classes, namespaces
  camelCase  | functions
  snake_case | variables

- source files use soft-indendation of four. To setup automatic
  code formatting of proposed changes using `clang-format`, install
  the provided git commit hook:

  ``` bash
  ./scripts/git-pre-commit-format install
  ```

### CODE REVIEW

File: src/space.h [hunk 1]
```diff
@@ -23,6 +23,7 @@ struct Change {
     bool all = false;        //!< Set to true if *everything* has changed
     bool dN = false;         //!< True if the number of atomic or molecular species has changed
     bool moved2moved = true; //!< If several groups are moved, should they interact with each other?
+    bool chargeMove = false;
```
Comment (mlund) [inline src/space.h]: Describe what this is doing

File: src/move.h [hunk 2]
```diff
@@ -283,6 +321,39 @@ class ChargeMove : public Movebase {
...
+/**
+ * @brief Transfers charge between two molecules
+ */
+class ChargeTransfer : public Movebase {
+  private:
+    typedef typename Space::Tpvec Tpvec;
+    Space &spc;           // Space to operate on
+    Average<double> msqd; // mean squared displacement
+    double dq = 0, deltaq = 0;
+    double sumCharges1 = 0;
```
[the remark this answers is not in the document]
Comment (mlund) [reply src/move.h]: btw. `changes` and `charges` are very easily confused. Consider renaming.

File: src/move.cpp [hunk 3]
```diff
@@ -387,6 +391,198 @@ ChargeMove::ChargeMove(Space &spc) : spc(spc) {
...
+                sumCharges2 = 0; 
+                sumChanges2 = 0; 
+                deltaq = dq * (slump() - 0.5);
+                changeQ1.clear();
+                changeQ2.clear();
+                cdata1.index = Faunus::distance(spc.groups.begin(), git1);
+                cdata2.index = Faunus::distance(spc.groups.begin(), git2);
+                for (i=0; i < numOfAtoms1; i++) {
+                    auto p = git1->begin()+i;
+                    //atomid = Faunus::distance(git2->begin(), p);
```
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.cpp]: fixed

File: src/move.cpp [hunk 4]
```diff
@@ -387,6 +391,198 @@ ChargeMove::ChargeMove(Space &spc) : spc(spc) {
...
+    _roundjson(j, 3);
+}
+void ChargeTransfer::_from_json(const json &j) {
+    try {
+        dq = j.at("dq").get<double>();
+        std::string molname1 = j.at("mol1");
+        std::string molname2 = j.at("mol2");
+        molrange1 = j.at("molrange1").get<std::vector<double>>();
+        molrange2 = j.at("molrange2").get<std::vector<double>>();
+        min1 = j.at("min1").get<std::vector<double>>();
```
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.cpp]: fixed

File: src/move.cpp [hunk 5]
```diff
@@ -583,6 +779,185 @@ TranslateRotate::TranslateRotate(Space &spc) : spc(spc) {
...
+        if (ref2 == atoms.end())
+            throw std::runtime_error("unknown reference atom '" + refname2 + "'");
+        molid = it->id();
+        refid1 = ref1->id();
+        refid2 = ref2->id();
+        dir = j.value("dir", Point(1, 1, 1));
+        dprot = j.at("dprot");
+        dptrans = j.at("dp");
+        p = j.at("p");
+        apad = j.at("apad");
```
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.cpp]: fixed

File: src/move.cpp [hunk 6]
```diff
@@ -181,6 +181,8 @@ Propagator::Propagator(const json &j, Space &spc, MPI::MPIController &mpi) {
             try {
                 if (it.key() == "moltransrot")
                     this->template push_back<Move::TranslateRotate>(spc);
+                else if (it.key() == "biasedmoltransrot")
```
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.cpp]: fixed

File: src/move.cpp [hunk 7]
```diff
@@ -583,6 +779,185 @@ TranslateRotate::TranslateRotate(Space &spc) : spc(spc) {
...
+        auto ref1 = slump.sample(reflist1.begin(), reflist1.end());
+        auto ref2 = slump.sample(reflist2.begin(), reflist2.end());
+        cylAxis = spc.geo.vdist( ref2->pos, ref1->pos )*0.5; // half vector between reference atoms
+        origo = ref2->pos - cylAxis;                         // coordinates of middle point between reference atoms: new origo
+        a = cylAxis.norm()+apad;                             // Length of a axis of sphere/ellipsoid
+
+        if (not it->empty()) {
+            assert(it->id == molid);
+
+            randNbr = slump();
```
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.cpp]: fixed

File: src/average.h [hunk 8]
```diff
@@ -77,6 +77,61 @@ namespace Faunus
       } // de-serialize from stream
   };
 
+      template<class T> class AverageExt
```
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/average.h]: removed, not needed

File: src/move.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.cpp]: fixed, renamed to rx and ry

File: src/move.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (vaspelin) [reply src/move.h]: fixed

File: src/space.h (hunk 1 above)
Comment (vaspelin) [reply src/space.h]: fixed

**Pull request merged**